crates/wasi-provider/src/states/starting.rs

```diff
@@ -111,22 +109,25 @@ impl State<PodState> for Starting {
 
         info!("Starting containers for pod {:?}", pod.name());
         for container in pod.containers() {
-            let container_handle = start_container(pod_state, &pod, &container).await?;
+            let container_handle = match start_container(pod_state, &pod, &container).await {
+                Ok(h) => h,
+                Err(e) => fail_fatal!(e),
```

Comment:
Same here re: fatal

Reply:
I originally had this transition to an error state, but found that caused a visible change in behaviour that caused the tests to fail.  I'd like to keep this as primarily a refactoring PR, and create a separate issue to identify and implement better error behaviour.